### ISSUE TITLE
Version 3.1 that added 10 second default timeout has bug.

### ISSUE THREAD

**Issue by RBornost** [Flagsmith/flagsmith-nodejs-client#132]
This PR https://github.com/Flagsmith/flagsmith-nodejs-client/pull/128/files#diff-f4f0d60a954687ddb23b817e910fec0c113105f276c9d4aba5866f3c511466ebL23

**The workaround is to just initialize the data.requestTimeoutSeconds to 10 seconds.**

The intended default of 10 seconds is never passed to the retryFetch which is subsequently expecting a value in milliseconds. My guess this was added because without a positive numeric value it would never timeout, which has also caused me grief I just didn't know it until i reviewed this code.

from [index.ts](https://github.com/Flagsmith/flagsmith-nodejs-client/blob/53f9f574f6675e64ceffc5683d50bd127371a341/sdk/index.ts#L90-L91)
These 2 lines
```        
this.requestTimeoutSeconds = data.requestTimeoutSeconds;
this.requestTimeoutMs = data.requestTimeoutSeconds ? data.requestTimeoutSeconds * 1000 : undefined;
```
         
 Should be changed to 
 ```       
this.requestTimeoutSeconds = data.requestTimeoutSeconds ?? this.requestTimeoutSeconds;
this.requestTimeoutMs = data.requestTimeoutSeconds ? data.requestTimeoutSeconds * 1000 : (this.requestTimeoutSeconds ? this.requestTimeoutSeconds * 1000 : undefined;)
```
   
   or possibly 

```        
this.requestTimeoutSeconds = data.requestTimeoutSeconds ?? this.requestTimeoutSeconds;
this.requestTimeoutMs = this.requestTimeoutSeconds ? this.requestTimeoutSeconds * 1000 : undefined;
```
       
The `this.requestTimeoutMs` is always undefined and when passed into the retry fetch it defaults it 10 milliseconds.
 
This is defined in [utils.ts](https://github.com/Flagsmith/flagsmith-nodejs-client/blob/53f9f574f6675e64ceffc5683d50bd127371a341/sdk/utils.ts#L19-L24)

```
export const retryFetch = (
    url: string,
    fetchOptions: RequestInit,
    retries: number = 3,
    timeout: number = 10// set an overall timeout for this function. <-------- this value is Milliseconds and should be 10 * 1000
):
```

and called from here in [index.ts](https://github.com/Flagsmith/flagsmith-nodejs-client/blob/53f9f574f6675e64ceffc5683d50bd127371a341/sdk/index.ts#L281-L291)

```       
const data = await retryFetch(
    url,
    {
        agent: this.agent,
        method: method,
        body: JSON.stringify(body),
        headers: headers
    },
    this.retries,
    this.requestTimeoutMs || undefined,
);
```

**Comment by matthewelwell** [comment]
@RBornost thanks for this, we'll take a look. 

Sidenote - I've updated the formatting of your description, please use 3 backticks in future to create code blocks rather than a single backtick which is for inline code. It's the difference between `this` and

```
this
which 
is 
multiline
```

**Comment by novakzaballa** [comment]
@RBornost this issue has been solved with the above commit and published with version 3.1.1 of this SDK.

**Comment by RBornost** [comment]
Excellent thank you for the quick fix.

And also regarding the formatting.. it was driving me nuts :)
